pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -132,60 +162,56 @@ def create_subscription(self,
             except exceptions.AlreadyExists:
                 pass
 
-        def subscribe(self,
-                      subscription: str,
-                      callback: Callable[[SubscriberMessage], Any],
-                      *,
-                      flow_control: Union[FlowControl, Tuple[int, ...]] = ()
-                      ) -> StreamingPullFuture:
+        async def pull(self, subscription: str, max_messages: int,
+                       *, session: Optional[Session] = None
+                       ) -> List[SubscriberMessage]:
             """
-            Create subscription through pubsub client, scheduling callbacks
-            on the event loop.
+            Pull messages from subscription
             """
-            sub_keepalive = self._subscriber.subscribe(
-                subscription,
-                self._wrap_callback(callback),
-                flow_control=flow_control)
-
-            self.loop.add_signal_handler(signal.SIGTERM, sub_keepalive.cancel)
-
-            return sub_keepalive
-
-        def run_forever(self, sub_keepalive: StreamingPullFuture) -> None:
+            url = f'{API_ROOT}/v1/{subscription}:pull'
+            headers = await self._headers()
+            body = {
+                'maxMessages': max_messages,
+            }
+            encoded = json.dumps(body).encode()
+            s = session or self.session
```

Comment:
Looks like a missing cast here, since `session` is an `aiohttp.ClientSession` but `self.session` is an AioSession. You probably want: `s = AioSession(session) if session else self.session`.

Reply:
Eagle eye!